### DISCUSSION TITLE
How to redirect mitmproxy log to my own logger?

### DISCUSSION THREAD

**Discussion by changsin** [mitmproxy/mitmproxy#6634] [General]
I am using DumpMaster to run mitmproxy.

    master_instance = dump.DumpMaster(
        opts,
        with_termlog=True,
        with_dumper=False,
    )
    master_instance.addons.add(my_addon)

 My problem is that I see that there are mitmproxy outputs like these:

[10:07:35.657][127.0.0.1:64469] client disconnect
[10:07:36.266][127.0.0.1:64467] Streaming response from edgeapi.slack.com.

They are output to stdout, but I would like to redirect them to my own logger which I get from     logger = logging.getLogger(mod_name)

Is there an easy way to do this?

**Comment by mhils** [comment]
I don't have an immediate answer for you, but you can check what we do for logging here:

https://github.com/mitmproxy/mitmproxy/blob/6bd19a77e2bb91274204420b8f211aff9eee0e31/mitmproxy/master.py#L37-L39
https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/addons/termlog.py
https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/log.py

**Comment by rosydawn6** [comment]
Termlog logs use print and will go to stdout: https://github.com/mitmproxy/mitmproxy/blob/6c7089f7a3bdc2f13017599fc9edfe247af65b67/mitmproxy/addons/termlog.py#L46

Depending on your specific configuration the root logger handler will look something like this:  [<TermLogHandler (DEBUG)>, <LegacyLogEvents (NOTSET)>, <ErrorCheckHandler (ERROR)> ].   Depending on your requirements you could redirect stdout from the mitmproxy/mitmdump command invocation to one of your log handlers which does not require a code change.  Or you could add a handler to the root logger, taking care of the propagate option, which would send termlog events to the handler or do other modifications to the root handlers.
